### DISCUSSION TITLE
Problem with changing host.

### DISCUSSION THREAD

**Discussion by xmyoard** [GriefPrevention/GriefPrevention#1605] [General]
Hello there!
I used aternos as the host for the server, now I changed the host to a premium host, so I need to copy the data also.
But aternos doesn't allow downloads and upload as their ToS so is there any way like any commands to copy or dump all claim data
to the web or somewhere with the help of a plugin? I can copy and paste each file because there are more than 200 claims, so is there a way, please let me know.

Warm regards,
~ Adarsh A

**Comment by bigpresh** [comment]
No, there's no commands to dump claim data like that - the claim data is stored under `plugins/GriefPreventionData` and it's expected that you can copy / archive / backup / snapshot that however you like.

Do they honestly not let you access your data? I'd try raising a support request with them?  Their site does claim "Automatic Backups -
We always keep a backup of your server, just in case you need it.".  In their console, at https://aternos.org/backups/ they suggest "Please connect your Google account to save backups in Google Drive." - you could try that?

**Reply by xmyoard** [reply]
No, there are backups made by aternos, but only Aternos's team can view or see the files, 
I asked the support, they said it can't be done or downloaded for copyright issues and or legal things :/
If that's a problem, can you add like /gp dump in which it uploads all the claim data to the web and we can download it as a .zip file and after a week it can be deleted? if its possible please add it :)

**Reply by bigpresh** [reply]
No, I don't think we'd be adding a very rarely used feature to compensate for the failings of a particular host, sorry.

**Reply by xmyoard** [reply]
ah, ok that's up to you, and it's not for a particular host, even if someone changes his mind from one premium server to another premium one?
it's not only for one host, and it's used for backing up the data or something.
anyways it's up to you guys.
and thanks for the response though

**Reply by bigpresh** [reply]
I've never encountered a host where you can't access your own data before, so I'd very much hope that's a rare thing.  Usually, you could just zip/tar up the contents of `plugins/GriefPreventionData` (or even all of `plugins`) and copy it across by whatever method you'd prefer, whether that's something in a host's web admin console if they provide one, or via SSH / SFTP / SCP / whatever.

I'd say it could be implemented by an addon, but it looks like Aternos only let you install a small subset of plugins they've approved, and if they don't want you to be able to access your data then I bet they'd not improve such a plugin.

Sorry I can't be more helpful!

**Reply by xmyoard** [reply]
yeah, thanks for letting me know.
as said do you know any plugins that can export these things, and for luck, aternos have MySQL support for plugins which means we will be able to export/connect to that?
and thanks for the help really appreciated :)

**Comment by Jikoo** [comment]
Most backup plugins offer offsite uploads via various services (FTP, DropBox, etc.). Configure one to back up GP's data and run a backup, voila.

**Reply by xmyoard** [reply]
thanks for the response.
aternos doesn't support FTP, Idk how to backup to dropbox and I don't think there is a way for this? you can give it a try by making an aternos account.

**Reply by Jikoo** [reply]
That's why you use a backup plugin and don't attempt to use their services for it. I assume you can install whatever plugin, yes? A [quick search](https://www.google.com/search?q=bukkit+dropbox+backup) yields a number of promising results. I've never used any of these (I always used to use the now-defunct SimpleBackup's FTP system) but I'd be inclined to check out [DriveBackupV2](https://github.com/MaxMaeder/DriveBackupV2).

**Reply by bigpresh** [reply]
> That's why you use a backup plugin and don't attempt to use their services for it. I assume you can install whatever plugin, yes?

From a quick play, it looks like Aternos only allow you to install a small subset of plugins they've approved - you pick from their list, you can't just upload & install what you want, and it doesn't look like they have any approved backup plugins (the cynical part of me says because that would make it easier to migrate your data out to another host...)

**Reply by Jikoo** [reply]
Shame. I guess the only alternative is to swap to MySQL and use an external DB, though unfortunately GP's MySQL integration is pretty terrible.

**Reply by xmyoard** [reply]
Hey, You all are correct, DriveBackupV2 is awesome, if it was on Spigot we can do that, and I saw a plugin named [BackupLikeMe](https://dev.bukkit.org/projects/backuplikeme) Its there on aternos, but it is for an old version.
Now, How wud I change from aternos to my other host.
argh!!! I am helpless!!
